src/transitions/cond/MCCondBroadcast.cpp

```diff
@@ -1,17 +1,94 @@
+      std::make_shared<mcmini::ConditionVariable>(shadow, std::move(policy));
+    // Register the mutex on the stack.
+    state->registerVisibleObjectWithSystemIdentity(condSystemId, newCond);
+    // condThatExists =
+    //   std::static_pointer_cast<MCConditionVariable, MCVisibleObject>(
+    //     newCond);
+    // Get the current cond var on the stack, along with condSystemIdentity
+    //   (the pthread_cond_t struct of the user).
+    condThatExists =
+      state->getVisibleObjectWithSystemIdentity<MCConditionVariable>(
+        condSystemId);
+    // This is still MCConditionVariable::Shadow::undefined;
+    // We set that in the next 'if' statement.
+  }
+
+  assert(condThatExists != nullptr);
+
+  if (testIfValidCond || condThatExists -> shadow.state ==
+                         MCConditionVariable::Shadow::undefined) {
+    // Then create and register an MCConditionVariable object.
+    pthread_cond_t cond_initializer = PTHREAD_COND_INITIALIZER;
+    // NOTE:  If PTHREAD_COND_INITIALIZER was used on a global variable,
+    //        the macro is all zeros, and the data segment is all zeros.
+    //        So, we can't tell if the initializer was or was not used.
+    //          But if a local variable of type pthread_cond_t is not
+    //        not initialized with PTHREAD_COND_INITIALIZER, and was
+    //        not initialized with pthread_cond_init, then this code can
+    //        truly catch an error.  pthread_cond_signal() would return
+    //        error "Illegal argument", but only if the user had checked
+    //        the return value of pthread_cond_signal, and then the errno.
+    pthread_cond_t local_cond;
+    struct iovec iov_remote_cond =
+      { .iov_base = condSystemId, .iov_len = sizeof(cond_initializer) };
+    struct iovec iov_local_cond =
+      { .iov_base = &local_cond, .iov_len = sizeof(cond_initializer) };
+    process_vm_readv(trace_pid, &iov_local_cond, 1, &iov_remote_cond, 1, 0);
+    if (memcmp(&local_cond, &cond_initializer, sizeof(cond_initializer)) != 0) {
+      mcprintf("\n*** ERROR (pthread_cond_broadcast/traceId: %d): The cond var\n"
+               "***     was never initialized.\n"
+               "***   Use pthread_cond_init() or PTHREAD_COND_INITIALIZER.\n"
+               "***   Additional McMini traces will not work with this bug.\n"
+               "***   McMini is exiting now.\n\n", traceId);
+      // No need to patch it with process_vm_writev.  We will stop now.
+      mc_stop_model_checking(EXIT_FAILURE);
+    }
+    if (condThatExists -> shadow.state ==
+        MCConditionVariable::Shadow::undefined) {
+      condThatExists -> shadow.state = MCConditionVariable::Shadow::initialized;
+    }
+  }
+
+  assert(condThatExists -> shadow.state ==
+         MCConditionVariable::Shadow::initialized);
+
   MC_REPORT_UNDEFINED_BEHAVIOR_ON_FAIL(
     condThatExists != nullptr, "Attempting to signal a condition "
                                "variable that is uninitialized");
```

Comment:
Likewise, this assert will no longer fire as we're producing the condition variable should McMini be otherwise unaware of it beforehand (i.e. no `pthread_cond_init()` call).